### ISSUE TITLE
could you please expose 3.2.3.1-SNAPSHOT to sonatype repository?

### ISSUE THREAD

**Issue by zifnab87** [orientechnologies/orientdb-gremlin#127]
Hi I am trying to bring make OrientDB to work again through spring-data (https://github.com/zifnab87/spring-data-gremlin/tree/TP3-before-merge) using Tinkerpop 3.2+. I am curious if it is possible that the 3.2.3.1-SNAPSHOT is exposed in a repo so I can get it easily. Thanks!

**Comment by maggiolo00** [comment]
hi @zifnab87 

we started to rolling a snapshot here

https://oss.sonatype.org/content/repositories/snapshots/com/orientechnologies/orientdb-gremlin/3.0.0-SNAPSHOT/

The version number of the driver has been aligned with OrientDB 3.x development and now supports TP3 3.2.4.

Do you need TP3 3.2.4 support for OrientDB 2.2.x ?

**Comment by zifnab87** [comment]
Thanks! I have seen that.. my main problem is that OrientDB-core still uses blueprints.Vertex etc instead of the latest Tinkerpop3

**Comment by maggiolo00** [comment]
@zifnab87 

yes in 2.2.x if you use the module `orientdb-graph`. Starting from 3.x we develop our own MultiModel API (Graph/Document) and the TP3 driver uses those API.

**Comment by zifnab87** [comment]
Oh I see so which dependencies do I need for 3.x so that Tinkerpop 3 is supported natively? Do you have a snapshot or something? Thanks

**Comment by maggiolo00** [comment]
hi @zifnab87 

yes we have a snapshot [here](https://oss.sonatype.org/content/repositories/snapshots/com/orientechnologies/orientdb-gremlin/3.0.0-SNAPSHOT/)


you can add it with

```
<dependency>
            <groupId>com.orientechnologies</groupId>
            <artifactId>orientdb-gremlin</artifactId>
            <version>3.0.0-SNAPSHOT</version>
  </dependency>
```

Please tell me if you need further help 
Thanks

**Comment by zifnab87** [comment]
Yeah I have used that already - I mean orientdb 3.x with native Tinkerpop 3.2.x support - do you have a snapshot for that? Thanks!

**Comment by maggiolo00** [comment]
Yeah we have a snapshot of OrientDB + Tp3 

https://oss.sonatype.org/content/repositories/snapshots/com/orientechnologies/orientdb-community-tp3/3.0.0-SNAPSHOT/

**Comment by zifnab87** [comment]
This is great! A few more questions.. any clue where I can find those - I couldn't find any repository of the code above so I can find those methods somewhere! Thanks!

http://orientdb.com/javadoc/latest/com/tinkerpop/blueprints/impls/orient/OrientBaseGraph.html#getVertexBaseType--
http://orientdb.com/javadoc/latest/com/tinkerpop/blueprints/impls/orient/OrientBaseGraph.html#getEdgeBaseType--
http://orientdb.com/javadoc/latest/com/tinkerpop/blueprints/impls/orient/OrientBaseGraph.html#getIndex-java.lang.String-java.lang.Class-
http://orientdb.com/javadoc/latest/com/tinkerpop/blueprints/impls/orient/OrientBaseGraph.html#getRawGraph--

**Comment by maggiolo00** [comment]
Hi @zifnab87 

in the new repo 

`OrientBaseGraph#getRawGraph` has been replaced by `OrientGraph#getRawDatabase`

For the other APIs i think there is no equivalent. You can use directly the API of`ODatabaseDocument` from  `OrientGraph#getRawDatabase`
